controllers/providers/kubernetes/crd.go

```diff
@@ -170,11 +172,33 @@ func ProcessCRDStrategy(kube dynamic.Interface, instanceGroup *v1alpha1.Instance
 
 	if strings.EqualFold(resourceStatus, successString) {
 		log.Info("custom resource succeeded", "instancegroup", instanceGroupNamespacedName, "resource", customResourceName, "status", resourceStatus)
+		status.SetStrategyRetryCount(0)
 		return true, nil
 	}
 
 	if strings.EqualFold(resourceStatus, failureString) {
 		log.Info("custom resource failed", "instancegroup", instanceGroupNamespacedName, "resource", customResourceName, "status", resourceStatus)
+		maxRetries := *strategy.MaxRetries
```

Comment:
One other option is to keep decrementing the strategy.MaxRetries until it reaches 0. Avoids using an extra variable...